### DISCUSSION TITLE
Sourcing alias from .bashrc for rm -rf

### DISCUSSION THREAD

**Discussion by kenderipa** [jarun/nnn#1960] [Q&A] [answered]
Hello,

I've decided to use trash system-wide, including for nnn. Trash management programs supported by nnn are way too heavy for me dependency-wise, so I'm not using them. Intead, I've decided to use a more lightweight program (rmw). I've aliased it in my .bashrc (alias rm=rmw), and it works everywhere, except for nnn. nnn ignores it and seemingly keeps using "true" rm -rf, completely bypassing my trash directory. Now I know that my .bashrc is sourced correctly, since NNN_FIFO, NNN_PLUG and NNN_OPENER work and can be seen in the help. Can I somehow source my rm alias for nnn too? Thanks!

**Comment by N-R-K** [comment]
> Can I somehow source my rm alias for nnn too?

Shell aliases are for interactive shell interactions. It doesn't get passed to other processes.

**Reply by kenderipa** [reply]
> > Can I somehow source my rm alias for nnn too?
> 
> Shell aliases are for interactive shell interactions. It doesn't get passed to other processes.

So there's no way right now? Would a feature request make sense?

**Reply by kenderipa** [reply]
> > > Can I somehow source my rm alias for nnn too?
> > 
> > 
> > Shell aliases are for interactive shell interactions. It doesn't get passed to other processes.
> 
> So there's no way right now? Would a feature request make sense?

Not for the alias sourcing, of course, but for the use of external remove binaries.

**Reply by 0xACE** [reply]
I dont use the trash functionality, so I can't tell what behavior you are expecting

but basically you would probably get away by replacing `sed 's/"rm\b/"rmw/g' nnn.c` if your `rmw` supports the same flags as `rm` after that just run `git diff` and process each line for potential issues.

**Reply by N-R-K** [reply]
> So there's no way right now? Would a feature request make sense?

Without editing the source code, not really (you can make a dummy wrapper script `trash-put` but maybe that's a bit hacky).

See https://github.com/jarun/nnn/discussions/1963#discussioncomment-11419296 for a proposed solution.

**Comment by N-R-K** [comment, marked as answer]
`master` branch now supports custom command in `$NNN_TRASH`. You don't need to patch the source or use some wrapper anymore. See the [wiki](https://github.com/jarun/nnn/wiki/Usage#configuration) and manpage for more info.

**Reply by kenderipa** [reply]
Thanks for ipmlementing it!
